src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1069,82 +1070,21 @@ class SyncFlowTypes(Enum):
-                                         http_status_code=500, component='rgw')
-        except SubprocessError as error:
-            raise DashboardException(error, http_status_code=500, component='rgw')
-
-        rgw_zonegroup_modify_cmd = ['zonegroup', 'modify',
-                                    '--rgw-realm', realm_name,
-                                    '--rgw-zonegroup', zonegroup_name]
-        if zonegroup_endpoints:
-            rgw_zonegroup_modify_cmd.append('--endpoints')
-            rgw_zonegroup_modify_cmd.append(zonegroup_endpoints)
-        rgw_zonegroup_modify_cmd.append('--master')
-        rgw_zonegroup_modify_cmd.append('--default')
-        try:
-            exit_code, _, err = mgr.send_rgwadmin_command(rgw_zonegroup_modify_cmd)
-            if exit_code > 0:
-                raise DashboardException(e=err, msg='Unable to modify zonegroup {}'.format(zonegroup_name),  # noqa E501  #pylint: disable=line-too-long
-                                         http_status_code=500, component='rgw')
-        except SubprocessError as error:
-            raise DashboardException(error, http_status_code=500, component='rgw')
-
-        rgw_zone_modify_cmd = ['zone', 'modify', '--rgw-realm', realm_name,
-                               '--rgw-zonegroup', zonegroup_name,
-                               '--rgw-zone', zone_name]
-        if zone_endpoints:
-            rgw_zone_modify_cmd.append('--endpoints')
-            rgw_zone_modify_cmd.append(zone_endpoints)
-        rgw_zone_modify_cmd.append('--master')
-        rgw_zone_modify_cmd.append('--default')
-        try:
-            exit_code, _, err = mgr.send_rgwadmin_command(rgw_zone_modify_cmd)
-            if exit_code > 0:
-                raise DashboardException(e=err, msg='Unable to modify zone',
-                                         http_status_code=500, component='rgw')
-        except SubprocessError as error:
-            raise DashboardException(error, http_status_code=500, component='rgw')
-
-        if access_key and secret_key:
-            rgw_zone_modify_cmd = ['zone', 'modify', '--rgw-zone', zone_name,
-                                   '--access-key', access_key, '--secret', secret_key]
-            try:
-                exit_code, _, err = mgr.send_rgwadmin_command(rgw_zone_modify_cmd)
-                if exit_code > 0:
-                    raise DashboardException(e=err, msg='Unable to modify zone',
-                                             http_status_code=500, component='rgw')
-            except SubprocessError as error:
-                raise DashboardException(error, http_status_code=500, component='rgw')
+class RgwMultisiteAutomation:
+    def __init__(self):
+        self.progress_id = str(uuid.uuid4())
+        self.progress_title = ''
+        self.progress_done = 0
+        self.progress_total = 2  # Total number of major steps
+
+    def update_progress(self, step_title):
```

Comment:
Although you are passing step but this is better param name
```suggestion
    def update_progress(self, progress_title):
```

Reply:
Done, Thanks